readthedocs/search/views.py

```diff
@@ -75,95 +116,88 @@ def get_serializer_context(self, project, version_slug):
         }
         return context
 
-    def get(self, request, project_slug=None):
-        request_type = None
-        use_advanced_query = True
-        if project_slug:
-            project_obj = self._get_project(project_slug)
-            use_advanced_query = not project_obj.has_feature(
-                Feature.DEFAULT_TO_FUZZY_SEARCH,
-            )
-            request_type = request.GET.get('type', 'file')
+    def get(self, request, project_slug):
+        project_obj = self._get_project(project_slug)
+        use_advanced_query = not project_obj.has_feature(
+            Feature.DEFAULT_TO_FUZZY_SEARCH,
+        )
 
-        version_slug = request.GET.get('version', LATEST)
+        user_input = UserInput(
+            query=request.GET.get('q'),
+            type='file',
+            project=project_slug,
+            version=request.GET.get('version', LATEST),
+            role_name=request.GET.get('role_name'),
+            language=None,
+        )
+
+        results, facets = self._search(
+            user_input=user_input,
+            use_advanced_query=use_advanced_query,
+        )
 
+        context = self.get_serializer_context(project_obj, user_input.version)
+        results = PageSearchSerializer(results, many=True, context=context).data
+
+        template_context = user_input._asdict()
+        template_context.update({
+            'results': results,
+            'facets': facets,
+            'project_obj': project_obj,
+        })
+
+        return render(
+            request,
+            'search/elastic_search.html',
+            template_context,
+        )
+
+
+class GlobalSearchView(SearchViewBase):
+
+    """
+    Global user search on the dashboard.
```

Comment:
This is confusing, what exactly is being searched? 

```suggestion
    Global search enabled for logged out users and anyone using the dashboard.
```